### ISSUE TITLE
MacOS + Graal AOT throws exception due to incorrect osArch value (amd64 instead of x86_64)

### ISSUE THREAD

**Issue by Songzhizong** [luben/zstd-jni#336]
#### Problem Description:

When running on **MacOS** with **Graal AOT**, the following exception occurs:

```
Unsupported OS/arch, cannot find /darwin/amd64/libzstd-jni-1.5.6-8.dylib or load zstd-jni-1.5.6-8 from system libraries.
```

This exception occurs because `System.getProperty("os.arch")` returns **`amd64`** instead of the expected **`x86_64`**, causing the system to be unable to find the corresponding **libzstd-jni** file.

#### Root Cause:

On MacOS, when running **Graal AOT**, **`os.arch`** is set to **`amd64`**, which prevents **Zstd JNI** from loading the correct `.dylib` file since the path depends on the `x86_64` architecture rather than `amd64`.

#### Possible Solution:

To fix this issue, it might be necessary to modify the implementation of the **`com.github.luben.zstd.util.Native.osArch()`** method. Specifically, when **`osName`** is **`darwin`** and **`osArch`** is **`amd64`**, it should return **`x86_64`**, allowing the system to find the correct library file.

#### Expected Behavior:

After the fix, **Graal AOT** on MacOS should be able to correctly load the **zstd-jni** library and no longer throw the related exception.

#### Environment Information:

- **Operating System**: MacOS
- **Architecture**: amd64 (returned value)
- **Tool**: Graal AOT
- **Zstd Version**: 1.5.6-8

**Comment by luben** [comment]
Yes, looks like the proposed work-around will work. Interesting though that the normal JVM returns `x86_64` but Graal returns `amd64`

**Comment by luben** [comment]
I pushed v1.5.6-9 with suggested workaround. Give it a try.

**Comment by Songzhizong** [comment]
v1.5.6-9 solves this problem
